src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -59,6 +59,31 @@ public static int getFileSizeInKB(String filePath) {
         return Math.toIntExact((file.length() / BYTES_PER_KILOBYTES) + 1L); // Add one so that integer division rounds up
     }
 
+    /**
+     * Retrieves a field mapping from properties map by recursively searching through the map.
+     *
+     * @param properties A map representing properties, where each key is a property name and
+     *                   the value is either a sub-map of properties or the property value itself.
+     * @param field      The name of the field to retrieve.
+     * @return           The value of the field if found, or null if the field is not present in the map.
+     */
+    public static Object getFieldMapping(Map<String, Object> properties, String field) {
+        if (properties.containsKey(field)) {
+            return properties.get(field);
+        } else {
+            for (Map.Entry<String, Object> entry : properties.entrySet()) {
+                Object value = entry.getValue();
+                if (value instanceof Map) {
+                    Object result = getFieldMapping((Map<String, Object>) value, field);
+                    if (result != null) {
+                        return result;
+                    }
+                }
+            }
+        }
+        return null;
```

Comment:
can we return Optional if we know null is a valid return type

Reply:
I talked with Martin offline on it, the Optional object will return nested optional object, which make it difficult to use in following steps.